lib/client/bespoke-client.ts

```diff
@@ -47,18 +49,33 @@ export class BespokeClient {
             }
         );
 
+
+
         // If the socket gets closed, probably server-side issue
         // We do not do anything in this case other than
         this.socketHandler.onCloseCallback = function() {
             if (!self.shuttingDown) {
                 LoggingHelper.error(Logger, "Socket closed by bst server: " + self.host + ":" + self.port);
+            }
+
+            if (self.reconnectRetries < BespokeClient.RECONNECT_MAX_RETRIES) {
+                self.reconnectRetries++;
+                LoggingHelper.error(Logger, "Attempting to reconnect in " + self.reconnectRetries + " seconds");
+                setTimeout(function () {
+                    self.attemptConnection();
+                }, self.reconnectRetries * 1000);
+            } else {
                 LoggingHelper.error(Logger, "Check your network settings - and try connecting again.");
                 LoggingHelper.error(Logger, "If the issue persists, contact us at Bespoken:");
                 LoggingHelper.error(Logger, "\thttps://gitter.im/bespoken/bst");
                 self.shutdown();
             }
         };
 
+        if (this.keepAlive) {
```

Comment:
Should this go inside a successful connection? Seems weird that we start it before we have connected

Reply:
we should kill the keep alive if we are attempting the connection, but should star